<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>586</width>
    <height>419</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <widget class="QTabWidget" name="tabWidget">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>581</width>
     <height>431</height>
    </rect>
   </property>
   <property name="currentIndex">
    <number>0</number>
   </property>
   <widget class="QWidget" name="tab_RSA">
    <attribute name="title">
     <string>RSA</string>
    </attribute>
    <widget class="QTabWidget" name="tabWidget_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>561</width>
       <height>381</height>
      </rect>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab_GenerateKey_RSA">
      <attribute name="title">
       <string>Generate Key</string>
      </attribute>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>10</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>p</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>40</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>q</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_p_rsa">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>10</y>
         <width>231</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_q_rsa">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>40</y>
         <width>231</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>70</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>e</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_e_rsa">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>70</y>
         <width>231</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_n_rsa">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>140</y>
         <width>231</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_16">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>140</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>n</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_17">
       <property name="geometry">
        <rect>
         <x>310</x>
         <y>10</y>
         <width>91</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Bilangan prima ke -</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_prima_ke_rsa">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>410</x>
         <y>10</y>
         <width>131</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_prima_rsa">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>310</x>
         <y>70</y>
         <width>231</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_generate_rsa">
       <property name="geometry">
        <rect>
         <x>189</x>
         <y>100</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Generate Key</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_get_prima_rsa">
       <property name="geometry">
        <rect>
         <x>460</x>
         <y>40</y>
         <width>80</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Get Prime</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_d_rsa">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>170</y>
         <width>231</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_18">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>170</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>d</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_save_private_key_rsa">
       <property name="geometry">
        <rect>
         <x>174</x>
         <y>220</y>
         <width>100</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Save Private Key</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_save_public_key_rsa">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>220</y>
         <width>100</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Save Public Key</string>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_enkripdekrip_RSA">
      <attribute name="title">
       <string>Enkripsi / Dekripsi</string>
      </attribute>
      <widget class="QLabel" name="label_15">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>30</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>n</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_12">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>10</y>
         <width>31</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>e / d</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_load_file_rsa">
       <property name="geometry">
        <rect>
         <x>460</x>
         <y>60</y>
         <width>80</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Load File</string>
       </property>
      </widget>
      <widget class="QTextEdit" name="textEdit_input_rsa">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>90</y>
         <width>530</width>
         <height>81</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_13">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Input</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_14">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>180</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Output</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_dekripsi_rsa">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>82</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Dekripsi</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_enkripsi_rsa">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>10</y>
         <width>82</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Enkripsi</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_ed_rsa">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>10</y>
         <width>231</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QTextEdit" name="textEdit_output_rsa">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>200</y>
         <width>530</width>
         <height>81</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_enkrip_dekrip_rsa">
       <property name="geometry">
        <rect>
         <x>200</x>
         <y>300</y>
         <width>151</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>Encrypt / Decrypt</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_save_file_rsa">
       <property name="geometry">
        <rect>
         <x>460</x>
         <y>290</y>
         <width>80</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Save File</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_n_ende_rsa">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>30</y>
         <width>231</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_load_key_rsa">
       <property name="geometry">
        <rect>
         <x>400</x>
         <y>10</y>
         <width>75</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Load Key</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_Elgamal">
    <attribute name="title">
     <string>Elgamal</string>
    </attribute>
    <widget class="QTabWidget" name="tabWidget_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>561</width>
       <height>381</height>
      </rect>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab_GenerateKey_Elgamal">
      <attribute name="title">
       <string>Generate Key</string>
      </attribute>
      <widget class="QLabel" name="label_19">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>10</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>p</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_20">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>40</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>g</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_p_elgamal">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>10</y>
         <width>231</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_g_elgamal">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>40</y>
         <width>231</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_21">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>70</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>x</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_x_elgamal">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>70</y>
         <width>231</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_y_elgamal">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>140</y>
         <width>231</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_22">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>140</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>y</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_generate_elgamal">
       <property name="geometry">
        <rect>
         <x>189</x>
         <y>100</y>
         <width>81</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Generate Key</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_save_private_key_elgamal">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>180</y>
         <width>100</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Save Private Key</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_save_public_key_elgamal">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>180</y>
         <width>100</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Save Public Key</string>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_enkripdekrip_RSA_2">
      <attribute name="title">
       <string>Enkripsi / Dekripsi</string>
      </attribute>
      <widget class="QLabel" name="label_25">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>50</y>
         <width>20</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>x,p</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_26">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>30</y>
         <width>31</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>y, g, p</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_load_file_elgamal">
       <property name="geometry">
        <rect>
         <x>460</x>
         <y>80</y>
         <width>80</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Load File</string>
       </property>
      </widget>
      <widget class="QTextEdit" name="textEdit_input_elgamal">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>110</y>
         <width>530</width>
         <height>81</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_27">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>90</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Input</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_28">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>200</y>
         <width>47</width>
         <height>13</height>
        </rect>
       </property>
       <property name="text">
        <string>Output</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_dekripsi_elgamal">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>82</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Dekripsi</string>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButton_enkripsi_elgamal">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>10</y>
         <width>82</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Enkripsi</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_ygp_elgamal">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>30</y>
         <width>231</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QTextEdit" name="textEdit_output_elgamal">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>220</y>
         <width>530</width>
         <height>81</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_enkrip_dekrip_elgamal">
       <property name="geometry">
        <rect>
         <x>200</x>
         <y>310</y>
         <width>151</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>Encrypt / Decrypt</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_save_file_elgamal">
       <property name="geometry">
        <rect>
         <x>460</x>
         <y>310</y>
         <width>80</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Save File</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_xp_elgamal">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>50</y>
         <width>231</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_load_pubkey_elgamal">
       <property name="geometry">
        <rect>
         <x>400</x>
         <y>30</y>
         <width>91</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Load Public Key</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_k_elgamal">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>10</y>
         <width>231</width>
         <height>20</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_29">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>10</y>
         <width>16</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>k</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_load_prikey_elgamal">
       <property name="geometry">
        <rect>
         <x>400</x>
         <y>50</y>
         <width>91</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Load Private Key</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_DiffieHelman">
    <attribute name="title">
     <string>Diffie-Hellman</string>
    </attribute>
    <widget class="QGroupBox" name="groupBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>331</width>
       <height>301</height>
      </rect>
     </property>
     <property name="title">
      <string>Generate Key</string>
     </property>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>n</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_n_dh">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>50</y>
        <width>291</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>g</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>x</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>110</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>y</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_g_dh">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>70</y>
        <width>291</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_x_dh">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>90</y>
        <width>291</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_y_dh">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>110</y>
        <width>291</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>200</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>K</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>230</y>
        <width>20</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>K'</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_generate_dh">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>140</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Generate</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>30</y>
        <width>47</width>
        <height>13</height>
       </rect>
      </property>
      <property name="text">
       <string>Input</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_K_dh">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>200</y>
        <width>291</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_Kdash_dh">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>230</y>
        <width>291</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>180</y>
        <width>47</width>
        <height>13</height>
       </rect>
      </property>
      <property name="text">
       <string>Output</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_save_key_dh">
      <property name="geometry">
       <rect>
        <x>239</x>
        <y>260</y>
        <width>81</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Save Key</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
